### PR TITLE
 Foreman 1.11 compatability (rails 4, et al.)

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: app/controllers/foreman_salt/concerns/hostgroups_controller_extensions.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module HostgroupsControllerExtensions
       end
 
       def salt_environment_selected
-        @hostgroup = Hostgroup.authorized(:view_hostgroups, Hostgroup).find_by_id(params[:hostgroup_id]) || Hostgroup.new(params[:hostgroup])
+        @hostgroup = Hostgroup.authorized(:view_hostgroups, Hostgroup).find_by(:id => params[:hostgroup_id]) || Hostgroup.new(params[:hostgroup])
```
Comment (dLobatog) [inline app/controllers/foreman_salt/concerns/hostgroups_controller_extensions.rb]: `find_by` methods ought to work on Rails 4. It's just `find_or_create_by` and `find_or_initialize_by` that were removed

**Comment by dLobatog** [comment]
@stbenjam For the record, I kicked this job to test this PR - http://ci.theforeman.org/job/test_plugin_pull_request/1963/ - if you don't have permissions in jenkins to rebuild it with the same parameters, ping me and I'll rerun it.

**Comment by stbenjam** [comment]
I do thanks, I already ran tests against this in http://ci.theforeman.org/job/test_plugin_pull_request/1960/, it's failing with some weird errors that are apparently also hitting foreman-tasks.  I haven't looked into it more.

**Comment by stbenjam** [comment]
https://github.com/theforeman/foreman-tasks/pull/162 will fix the test failures

**Comment by stbenjam** [comment]
This is becoming a PR to fix all the problems with foreman_salt, which is still very much a WIP.  
There's a lot of broken things because of reports STI changes, smart proxy page changes, etc over the last few months

The most recent fix tries to make foreman_salt a little more stable by introducing at least an extension point for the smart proxy buttons. That goes with https://github.com/theforeman/foreman/pull/3032

**Comment by stbenjam** [comment]
http://ci.theforeman.org/view/Plugins/job/test_plugin_pull_request/2010/ is mostly :green_heart:, but 1.9.3 is red because of https://github.com/theforeman/foreman/pull/3034. 

Should be green once that's merged.

**Comment by stbenjam** [comment]
Green http://ci.theforeman.org/view/Plugins/job/test_plugin_pull_request/2182/

**Pull request merged**